CMakeLists.txt

```diff
@@ -1,71 +1,52 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
-
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
-
-# TODO: fix the macOS bundle stuff
-if (POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
-endif()
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-include(FunctionReadVersion)
-
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_BUGFIX VER_BUILD VER_STATUS)
-# message(STATUS "Project version is: ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD} (${VER_STATUS})")
-
-project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
-
-# check for invalid compiler version/CXX standard as early as possible
-include(FunctionQbtCXXCompilerAndModeCheck)
-qbt_minimum_cxx_mode_check(14)
-message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
-            "Make sure libtorrent was built with the same C++ mode for ABI compatibility.")
-
-set(PROJECT_VERSION "${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}")
-
-if (NOT VER_BUILD EQUAL 0)
-    set(PROJECT_VERSION "${PROJECT_VERSION}.${VER_BUILD}")
-endif()
 
-set(PROJECT_VERSION "${PROJECT_VERSION}${VER_STATUS}")
-
-add_definitions(-DQBT_VERSION_MAJOR=${VER_MAJOR})
-add_definitions(-DQBT_VERSION_MINOR=${VER_MINOR})
-add_definitions(-DQBT_VERSION_BUGFIX=${VER_BUGFIX})
-add_definitions(-DQBT_VERSION_BUILD=${VER_BUILD})
-
-add_definitions(-DQBT_VERSION="v${PROJECT_VERSION}")
-add_definitions(-DQBT_VERSION_2="${PROJECT_VERSION}")
-
-include(GNUInstallDirs)
-include(FeatureSummary)
+project(qBittorrent
+    VERSION 4.3.0.0
```

Comment:
@glassez 

Isn't this meant to replace qmake in the somewhat near future? I made this change because I didn't think it was a good idea to always read this from the `.pri` and therefore have the CMake buildsystem implicitly depending on the qmake one. In fact, it's not even a good idea to pass the project version as a compile definition, since it causes recompilation of everything. Having it defined this way is the first step towards a minor version number-related refactor I have planned for the future that will solve this problem.

In the meantime, it's one more number to change in commits that increment the version - not much more work, and I'll make sure to nag about it if anyone forgets.

Reply:
>Isn't this meant to replace qmake in the somewhat near future? 

I think it's more about a few years.
Most likely, we should make both systems equally "official" with the start of Qt6 support. So I would not want edit several different files to change the project version, which can cause a lack of synchronization.

Reply:
@glassez 

> So I would not want edit several different files to change the project version

This is simply the price to pay for maintaining 2 different build systems like this. But it's not a very high one - it's just a single additional line change, and not very often. We're much worse off if we couple parts of one buildsystem's logic to the other IMO.

>I think it's more about a few years.
Most likely, we should make both systems equally "official" with the start of Qt6 support.

Oh god...

Reply:
>This is simply the price to pay for maintaining 2 different build systems like this. But it's not a very high one - it's just a single additional line change, and not very often

Okay, it's not that much of a problem for me if others agree with you.